fuuk/people/admin/options.py

```diff
@@ -147,39 +149,118 @@ class AgencyAdmin(TranslationAdmin):
+        """
+        return True
 
     def has_change_permission(self, request, obj=None):
         """
-        If `obj` is None, this should return True if the given request has
-        permission to delete *any* object of the given type.
+        The change permission has user, which is the editor, author or collaborator or has change permission for grants.
+        If `obj` is None, this should return True if the given request has permission to change *any* grant.
         """
-        if request.user.is_superuser:
-            return True
+        if obj:
+            if request.user == obj.editor or \
+                    request.user == getattr(obj.investigator_human, 'user', None):
+                return True
+            if obj.collaborators.filter(human__user=request.user).exists():
+                return True
+            return super(GrantAdmin, self).has_change_permission(request, obj)
+
+        return True
 
+    def has_delete_permission(self, request, obj=None):
+        """
+        The delete permission has user, which is the editor, author or collaborator or has delete permission for grants.
+        If `obj` is None, this should return True if the given request has permission to delete *any* grant.
+        """
         if obj:
-            if request.user == getattr(obj.author.human, 'user', None):
+            if request.user == obj.editor or \
+                    request.user == getattr(obj.investigator_human, 'user', None):
                 return True
-            for author in obj.co_authors.all():
-                if request.user == getattr(author.human, 'user', None):
-                    return True
-            return False
+            if obj.collaborators.filter(human__user=request.user).exists():
+                return True
+            return super(GrantAdmin, self).has_delete_permission(request, obj)
 
-        return super(GrantAdmin, self).has_change_permission(request, obj)
+        return True
 
     def get_queryset(self, request):
+        """
+        User with change permission can view all grants, other users can view only grants, where they are editor,
+        author or collaborator.
+        """
         queryset = super(GrantAdmin, self).get_queryset(request)
-        if request.user.is_superuser:
+        opts = self.opts
+        codename = get_permission_codename('change', opts)
+        if request.user.has_perm("%s.%s" % (opts.app_label, codename)):
             return queryset
-        human = request.user.human
-        return queryset.filter(
-            pk__in=queryset.filter(author__human=human) | queryset.filter(co_authors__human=human)
-        )
+        human = getattr(request.user, 'human', None)
+        if human:
+            return queryset.filter(
+                pk__in=(queryset.filter(investigator_human=human) | queryset.filter(collaborators__human=human) |
+                        queryset.filter(editor=request.user))
+            )
```

Comment:
Corrected using `Q` objects because I found out that using `|` operator on QuerySet is undocumented.